### DISCUSSION TITLE
Does mi.load_dict support another dictionary as one of the values?

### DISCUSSION THREAD

**Discussion by Theo-Wu** [mitsuba-renderer/mitsuba3#498] [Q&A]
## Summary

When I tried to use mi.load_dict to load a python dictionary that contains another dictionary as a value, an error occurred: ​[xml_v.cpp:127] Missing key 'type' in the dictionary.

## System configuration

  OS: Ubuntu 20.04.4 LTS
  CPU: Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz
  GPU: NVIDIA GeForce GTX 1660 SUPER
  Python: 3.9.7 (default, Sep 16 2021, 13:09:58) [GCC 7.5.0]
  NVidia driver: 511.65
  CUDA: 11.6.124
  LLVM: 12.0.0

  Dr.Jit: 0.2.2
  Mitsuba: 3.0.2
     Is custom build? False
     Compiled with: GNU 10.2.1
     Variants:
        scalar_rgb
        scalar_spectral
        cuda_ad_rgb
        llvm_ad_rgb

## Description

I was customizing a python plugin for my bsdf, and I wanted to pass a dictionary as one of the properties.

## Steps to reproduce

```
b = {
    'prop1':1,
    'a':{'prop2':2}
}
mi.load_dict(b)
```

**Comment by njroussel** [comment]
Hi @Theo-Wu 

Yes, _sort of._

If you take a look at some of the existing plugins examples, like [here](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#smooth-diffuse-material-diffuse), the dictionary can have nested dictionaries.
However, every single dictionary level should have at least one key `type`, which defines the plugin type. So, effectively, you can **only** have nested plugins and not a "raw" dictionnary as a plugin property/parameter.

Does this make sense?
